pele_platform/Utilities/Helpers/smiles_constraints.py

```diff
@@ -0,0 +1,92 @@
+from dataclasses import dataclass
+import re
+import pele_platform.Errors.custom_errors as ce
+
+@dataclass
+class SmilesConstraints:
+
+    input_pdb: str
+    constrain_core: str
+    resname: str
+    chain: str
+    spring_constant: float = 50.0
+
+    def run(self):
+        from rdkit import Chem
+        self._convert_to_smarts()
+        self._extract_ligand()
+        self._get_matches()
+        if self.substructures:
+            self._build_constraints()
+            return self.constraints
+        else:
+            raise ce.SubstructureError("Could not recognise the substructure specified in 'constrain_core'. This might be due to differences in ionisation states. Check, if the charges are identical and adjust your pattern, if necessary.\nCore SMARTS: {}\nLigand SMARTS: {}".format(self.smarts, Chem.MolToSmarts(self.ligand)))
```

Comment:
No flexibility when running this class. Let's try to aim that the user can run the methods separately as well. Like:

pdb = self._extract_ligand(self.input_pdb)